lib/manager/bundler/extract.ts

```diff
@@ -4,7 +4,59 @@ import { PackageFile, PackageDependency } from '../common';
 import { platform } from '../../platform';
 import { regEx } from '../../util/regex';
 import { extractLockFileEntries } from './locked-version';
-import { DATASOURCE_RUBYGEMS } from '../../constants/data-binary-source';
+import {
+  DATASOURCE_RUBYGEMS,
+  DATASOURCE_GITHUB,
+} from '../../constants/data-binary-source';
+
+function parseHashArgs(str: string): Partial<PackageDependency> {
+  const result: Partial<PackageDependency> = {};
+
+  const parsed: Record<string, string> = {
+    git: null,
+    github: null,
+    tag: null,
+  };
+
+  for (const key of Object.keys(parsed)) {
+    const regex = new RegExp(
+      `,\\s*(:${key}\\s*=>|${key}\\s*:)\\s*(['"])(?<value>[^'"]+)\\2`
```

Comment:
While named regex groups are optional, it's better to use numbered backreferences for quote characters, instead of code them separately.